### DISCUSSION TITLE
Getting error with grid_stat

### DISCUSSION THREAD

**Discussion by pouwereou** [dtcenter/METplus#2804] [File I/O] [answered]
Hello everyone,
This is my first post here and I would like to thank all the members of this community.
I am struggling to run the grid_stat using wrf output in grib2 format and the GPMIMERG data in netcdf format. This is the error I get: 
ERROR : MetNcFile::data(NcVar *, const LongArray &, DataPlane &) const -> needed 3 arguments for variable APCP, got 2.
I have tried several tricks, looked in the manual and the FAQ, but still have the same error. It's my first time using MET-11.1.1 and I need some help to make it work. I have attached is the data I used and also the config file via the anonymous FTP site.

**Comment by CPKalb** [comment, marked as answer]
Hi:

I am not able to find your data on our FTP server.  I see a folder labeled nimon_data, but I don't see any files inside that folder.  Could you try uploading the data again or perhaps posting your configuration file, and the header of the netCDF file here?

Based on the information provided above, it looks like the error is occurring while reading in the netCDF data.  How is your level specified in the configuration file, and what does the header of the netCDF file look like?

Thanks,
Christina

**Reply by pouwereou** [reply]
Hi Christina,
Sorry about that.
I just uploaded the files. I also uploaded a file called daily_precip_24.nc which is obtained by using pcp_combine (-subtract). When I use this too, the grid_stat does not work.

Thanks very much,
Pouwereou

**Reply by CPKalb** [reply]
Hi:

In your configuration file (GridStatConfig), it looks like you have set:
```
obs = fcst;
```

Although the forecast and observation data have the same variable name, they need to be specified differently since one is in grib format and the other is netCDF.  I think you need to change that line to say the following:

```
obs = {

   field = [
      {
        name       = "APCP";
        level      = [ "(0,*,*)" ];
        cat_thresh = [ >1.0, >=5.0 ];
      }
   ];

};
```

This ran successfully for me using the following command with the original netCDF file (GPMIMERG_20240901_2.nc):
```
grid_stat WRFPRS_d01.24 daily_precip_24.nc GridStatConfig
```
Sections [5.2.11](https://met.readthedocs.io/en/latest/Users_Guide/config_options.html#fcst) and [5.2.12](https://met.readthedocs.io/en/latest/Users_Guide/config_options.html#obs) of the MET documentation explains in more detail how to specify the field name and level for variables in netCDF and grib files

Does this work on your end?

Christina

**Reply by pouwereou** [reply]
Hi Christina,
I changed the configuration file as you said and the command works perfectly. Thank you very much for your help.

**Reply by pouwereou** [reply]
Hi,
Please I have another question. Suppose I want to evaluate my wrf simulation from 20240901_000000 to 20240910_000000. Should I run the grid_stat on each wrfoutput file that has been passed through pcp_combine (e.g. wrfout_20240901.nc, wrfout_20240902.nc ...) or should I merge them before running the grid_stat.

**Reply by CPKalb** [reply]
Hi:

To evaluate the whole WRF simulation, you would want to run Grid-Stat on each WRF output file (e.g. wrfout_20240901.nc, wrfout_20240902.nc ...) comparing to the observations at the correct time.  You will then get statistics for each time of the WRF simulation.  These statistics can be aggregated using the [Stat-Analysis tool](https://met.readthedocs.io/en/latest/Users_Guide/stat-analysis.html) to produce statistics over the entire WRF run.

Our METplus wrappers are designed to handle the workflow (looping) of running Grid-Stat over each WRF output file.  I'm including some links below on how to use the METplus wrappers to run Grid-Stat over multiple files that will hopefully help set this up.

[Grid-Stat basic use case with METplus wrappers](https://metplus.readthedocs.io/en/latest/generated/met_tool_wrapper/GridStat/GridStat.html#sphx-glr-generated-met-tool-wrapper-gridstat-gridstat-py)
[METplus documentation](https://metplus.readthedocs.io/en/latest/)

To run METplus, you will need access to python, with more specifics listed [here](https://metplus.readthedocs.io/en/latest/Users_Guide/overview.html#metplus-components-python-requirements).  It can be run on a machine with an [existing build](https://dtcenter.org/community-code/metplus/metplus-5-1-existing-builds) of METplus or by checking out the latest repository.  For METplus, configuration options are specified inside the METplus .conf file rather than directly through the MET configuration file.  Let me know if you have questions on this.

Christina

**Reply by pouwereou** [reply]
Hi Christina
Thank you for your reply, it is very clear to me. I will work on it and let you know if I have any problems.

Pouwereou

**Reply by CPKalb** [reply]
okay, great!  I'm going to mark this discussion as answered then and lock it to prevent future posts.  If you have questions on the setup for running multiple files, feel free to start a new discussion.

Christina
